src/lib/Preview.svelte

```diff
@@ -3,36 +3,49 @@
 
   export type CSS = Record<string, string>;
 
-  type ScenarioSize = { width?: string; height?: string; children?: number };
-  export type Scenario<C extends SvelteComponent = SvelteComponent> = {
+  type ScenarioSize = { width?: string; height?: string };
+  export type Scenario<
+    C extends SvelteComponent = SvelteComponent,
+    S = unknown,
+  > = {
     props: ComponentProps<C>;
+    slotData?: S;
     css?: CSS;
     size?: ScenarioSize;
   };
-  export type Scenarios<C extends SvelteComponent> = {
-    [key: string]: Scenario<C>;
+  export type Scenarios<C extends SvelteComponent, S = unknown> = {
+    [key: string]: Scenario<C, S>;
   };
 
   export type ScenarioState = Required<Scenario> & {
     size: Required<ScenarioSize>;
   };
 </script>
 
-<script lang="ts" generics="_C extends SvelteComponent">
+<script lang="ts" generics="_C extends SvelteComponent, _S">
   import { onMount, type ComponentType, type ComponentEvents } from "svelte";
   import ScenarioView from "$lib/ScenarioView.svelte";
 
   // _C is defined in the `generics` attribute of the `script` tag
   // but this is not recognized by eslint
   type C = _C; // eslint-disable-line no-undef
+  type S = _S; // eslint-disable-line no-undef
 
   export let component: ComponentType<C>;
-  export let scenarios: Scenarios<C>;
-  export let defaultCss: CSS = {};
```

Comment:
why is `defaultCss` being removed? Is there any other way now to set CSS values for scenarios without explicit css styles?

Reply:
`scenario.css` is required only on one [place in code](https://github.com/mrh1997/preview.svelte/blob/3e37efb4e222931b653bf79c22ecaa75c87bb28e/src/lib/ScenarioInstance.svelte#L54-L54). By writing `scenario.css ?? {}` there I could get rid of the ScenarioState and CSS types.

Reply:
`defaultCss`  was a property of the Preview component.
So as a user I was able to define a set css styles applied to all scenarios, this wont be possible now

Reply:
Just double checked, I use the `defaultCss` prop in my code only once.
If we remove it, I can workaround pretty easy, so I am fine with deleting this property.

Reply:
Oh. I missed the point! I didn't realize that this was an external property that allowed to set defaults. sorry...